bundles/pubsub/pubsub_protocol_wire_v1/src/pubsub_wire_protocol_impl.c

```diff
@@ -65,19 +65,40 @@ celix_status_t pubsubProtocol_destroy(pubsub_protocol_wire_v1_t* protocol) {
     return status;
 }
 
+celix_status_t pubsubProtocol_getHeaderSize(void* handle, size_t *length) {
+    *length = sizeof(int) * 5 + sizeof(short) * 2; // header + sync + version = 24
+
+    return CELIX_SUCCESS;
+}
+
+celix_status_t pubsubProtocol_getHeaderBufferSize(void* handle, size_t *length) {
+    return pubsubProtocol_getHeaderSize(handle, length);
+}
+
+celix_status_t pubsubProtocol_getSyncHeaderSize(void* handle,  size_t *length) {
+    *length = sizeof(int);
+    return CELIX_SUCCESS;
+}
+
 celix_status_t pubsubProtocol_getSyncHeader(void* handle, void *syncHeader) {
-    for (int i = 0; i < 5; ++i) {
-        ((char *) syncHeader)[i] = '\0';
-    }
     writeInt(syncHeader, 0, PROTOCOL_WIRE_SYNC);
-
     return CELIX_SUCCESS;
 }
 
+celix_status_t pubsubProtocol_isMessageSegmentationSupported(void* handle, bool *isSupported) {
+    *isSupported = false;
+    return CELIX_SUCCESS;
+}
 celix_status_t pubsubProtocol_encodeHeader(void *handle, pubsub_protocol_message_t *message, void **outBuffer, size_t *outLength) {
     celix_status_t status = CELIX_SUCCESS;
+    // Get HeaderSize
+    size_t headerSize = 0;
+    pubsubProtocol_getHeaderSize(handle, &headerSize);
```

Comment:
I have added extra comment in the protocol header file. The header size is used by the receiver to know how big the header is. The receiver needs to read the header with payload size / metedata size to know how big the message is. For static end points (external systems that don't use celix pubsub) we need to be able send / receive headerless message with only a payload. Then header is part of the payload. This means that the headerbuffer is not used because the header is part of the payload. For headerless data the user needs to create a dedicated protocol service. TCPadmin is the only pubsub admin that can be used as a static end point because together with the protocol admin it can define what's send over the line. ZMQ does have internal headers send in front of the data for ZMQ protocol handling